### PR TITLE
tests: gitignore addition

### PR DESCRIPTION
**Pull request by williamcroberts**
Only worry about shell scripts, unit tests, test fixtures,
the readme and the gitignore file itself.  and ignore the
test artifacts that end up in this directory.

Signed-off-by: William Roberts <william.c.roberts@intel.com>

### CODE REVIEW

File: test/.gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Ignore everything
+*
+
+# ...even if they are in subdirectories
+!*/
```
Comment (gwei3) [inline test/.gitignore]: I think you are trying to ignore anything in subdir, so shouldn't it be as below?
*/

**Comment by gwei3** [comment]
@williamcroberts looks like this need to be redo after your test cleanup changes merged.

File: test/.gitignore (hunk 1 above)
Comment (williamcroberts) [inline test/.gitignore]: Actually the * would get subdirs as well, so we can just delete the line and the incorrect comment.

**Comment by williamcroberts** [comment]
gwei3 this is OK to pull now.

**Pull request merged**